src/sw/redis++/subscriber.cpp

```diff
@@ -117,14 +108,27 @@ Subscriber::MsgType Subscriber::_msg_type(redisReply *reply) const {
         throw ProtoError("Null type reply.");
     }
 
-    auto type = reply::parse<std::string>(*reply);
-
-    auto iter = _msg_type_index.find(type);
-    if (iter == _msg_type_index.end()) {
-        throw ProtoError("Invalid message type.");
-    }
+    return _msg_type(reply::parse<std::string>(*reply));
+}
 
-    return iter->second;
+Subscriber::MsgType Subscriber::_msg_type(std::string const& type) const
```

Comment:
Adoption of such a scheme shouldn't decrease performance compared to the static map solution and provide a static initialization/destruction order safe alternative. 

If however there are concerns about performance, a string based lookup table can be definitely faster than both. Made an example implementation here http://coliru.stacked-crooked.com/a/0eccb19dbc800885 . The catch is that message descriptions ("subscribe", "unsubscribe" etc) have to hash differently which currently they do.